### PR TITLE
fix(connector_onboarding): Check if connector exists for the merchant account and add reset tracking id API

### PR DESCRIPTION
**Pull request by ThisIsMani**
## Type of Change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

1. This PR adds checks in connector_onboarding to only proceed if the connector exists for the given merchant_account.
2. Reset tracking id api has been added to connector onboarding, which will reset the tracking details of the onboarded merchant. This can be used to update the process without any interference of previous account.

### Additional Changes

- [x] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context

1. To catch any errors without the connector id check beforehand.
2. To make the update flow of connector onboarding better.

## How did you test it?

Postman.
```
curl --location 'http://localhost:8080/connector_onboarding/action_url' \
--header 'Content-Type: application/json' \
--header 'Authorization: Bearer JWT' \
--data '{
    "connector": "paypal",
    "connector_id": "",
    "return_url": "https://google.com"
}'
```
```
curl --location 'http://localhost:8080/connector_onboarding/sync' \
--header 'Content-Type: application/json' \
--header 'Authorization: Bearer JWT' \
--data '{
    "connector": "paypal",
    "profile_id": "",
    "connector_id": ""
}'
```
If the connector_id is invalid, these apis will throw this error
```
{
    "error": {
        "type": "invalid_request",
        "message": "Merchant connector account does not exist in our records",
        "code": "HE_02",
        "reason": " does not exist"
    }
}
```

This api will reset the progress of the onboarding flow. This api will give 200 OK if everything works correctly.
```
curl --location 'http://localhost:8080/connector_onboarding/reset_tracking_id' \
--header 'Content-Type: application/json' \
--header 'Authorization: Bearer JWT' \
--data '{
    "connector_id": "",
    "connector": "paypal"
}'
```

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Review by apoorvdixit88 — DISMISSED** [review]

File: crates/router/src/utils/connector_onboarding.rs [hunk 1]
```diff
@@ -34,3 +39,98 @@ pub fn is_enabled(
...
+    merchant_id: &str,
+) -> RouterResult<()> {
+    let key_store = state
+        .store
+        .get_merchant_key_store_by_merchant_id(
+            merchant_id,
+            &state.store.get_master_key().to_vec().into(),
+        )
+        .await
+        .change_context(ApiErrorResponse::InternalServerError)?;
```
Comment (Narayanbhat166) [inline crates/router/src/utils/connector_onboarding.rs:55]: why directly changing to `InternalServerError`. Use `not_found_response`

File: crates/router/src/utils/connector_onboarding.rs [hunk 2]
```diff
@@ -34,3 +39,98 @@ pub fn is_enabled(
...
+    state: &AppState,
+    connector_id: &str,
+    connector: enums::Connector,
+) -> RouterResult<()> {
+    let timestamp = common_utils::date_time::now_unix_timestamp().to_string();
+    if state
+        .store
+        .find_config_by_key(&build_key(connector_id, connector))
+        .await
+        .is_ok()
```
Comment (Narayanbhat166) [inline crates/router/src/utils/connector_onboarding.rs:82]: do not ignore errors, atleast log it

File: crates/router/src/utils/connector_onboarding.rs [hunk 3]
```diff
@@ -34,3 +39,98 @@ pub fn is_enabled(
...
+        state
+            .store
+            .update_config_by_key(
+                &build_key(connector_id, connector),
+                ConfigUpdate::Update {
+                    config: Some(timestamp),
+                },
+            )
+            .await
+            .change_context(ApiErrorResponse::InternalServerError)
```
Comment (Narayanbhat166) [inline crates/router/src/utils/connector_onboarding.rs:93]: use to `not_found_response`

**Review by Narayanbhat166 — DISMISSED** [review]

File: crates/router/src/utils/connector_onboarding.rs (hunk 3 above)
Comment (ThisIsMani) [reply crates/router/src/utils/connector_onboarding.rs:93]: We will enter this only if found it in the `if` condition.

**Pull request merged**